src/cmd/linuxkit/azure.go

```diff
@@ -0,0 +1,372 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+
+	"path/filepath"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	simpleStorage "github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+var (
+	simpleStorageClient     simpleStorage.Client
+	groupsClient            resources.GroupsClient
+	accountsClient          storage.AccountsClient
+	virtualNetworksClient   network.VirtualNetworksClient
+	subnetsClient           network.SubnetsClient
+	publicIPAddressesClient network.PublicIPAddressesClient
+	interfacesClient        network.InterfacesClient
+	virtualMachinesClient   compute.VirtualMachinesClient
+)
+
+func initializeAzureClients(subscriptionID, tenantID, clientID, clientSecret string) {
+	oAuthConfig, err := adal.NewOAuthConfig(azure.PublicCloud.ActiveDirectoryEndpoint, tenantID)
+	if err != nil {
+		fmt.Printf(err.Error())
+		log.Fatalf("Cannot get oAuth configuration")
+	}
+
+	token, err := adal.NewServicePrincipalToken(*oAuthConfig, clientID, clientSecret, azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		fmt.Printf(err.Error())
+		log.Fatalf("Cannot get service principal token")
+	}
+
+	groupsClient = resources.NewGroupsClient(subscriptionID)
+	groupsClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	accountsClient = storage.NewAccountsClient(subscriptionID)
+	accountsClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	virtualNetworksClient = network.NewVirtualNetworksClient(subscriptionID)
+	virtualNetworksClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	subnetsClient = network.NewSubnetsClient(subscriptionID)
+	subnetsClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	publicIPAddressesClient = network.NewPublicIPAddressesClient(subscriptionID)
+	publicIPAddressesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	interfacesClient = network.NewInterfacesClient(subscriptionID)
+	interfacesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+	virtualMachinesClient = compute.NewVirtualMachinesClient(subscriptionID)
+	virtualMachinesClient.Authorizer = autorest.NewBearerAuthorizer(token)
+
+}
+
+func getOrCreateResourceGroup(resourceGroupName, location string) *resources.Group {
+	var resourceGroup resources.Group
+	resourceGroup, err := groupsClient.Get(resourceGroupName)
+	if err != nil {
+		log.Fatalf("\nError in getting resource group")
```

Comment:
there shouldn't be a `\n` at the start of there error message.

Would it make sense to also print the error?